cdap-ui/app/features/admin/controllers/DatasetsController.js

```diff
@@ -1,9 +1,24 @@
 angular.module(PKG.name + '.feature.admin')
-  .controller('AdminDatasetsController', function ($scope) {
-    $scope.datasets = [{
-      name: 'STU',
-      type: 'Stream',
-      description: 'little miss muffet',
-      apps: '2'
-    }];
+  .controller('AdminDatasetsController', function ($scope, MyDataSource, $stateParams) {
+
+    var dataSrc = new MyDataSource($scope);
+
+    $scope.datasets = [];
+    $scope.streams = [];
+
+
+    dataSrc.request({
+      _cdapPath: '/namespaces/' + $stateParams.nsadmin + '/data/datasets'
+    })
+      .then(function(res) {
+        $scope.datasets = res;
+      });
+
+    dataSrc.request({
+      _cdapPath: '/namespaces/' + $stateParams.nsadmin + '/streams'
+    }, function(res) {
+      if (angular.isArray(res) && res.length) {
```

Comment:
Any specific reason why we are having this check here alone and not in datasets? Do we need this check?